app/models/order.rb

```diff
@@ -0,0 +1,68 @@
+class Order < ApplicationRecord
+  include ActiveModel::Dirty
+
+  define_attribute_methods
+
+  has_many :line_items, dependent: :destroy, after_add: :calculate_price_and_discount_price_on_add, after_remove: :calculate_price_and_discount_price_on_remove, before_add: :prevent_multiple_purchase_of_same_deal, autosave: true
+  belongs_to :user
```

Comment:
let's keep all belogs_to together and has_many together

Reply:
Done